src/js/angular/rest/mappers/import-mapper.js

```diff
@@ -63,8 +71,30 @@ const addResourceToTree = (root, resource) => {
         importServerResource.path = path.join('/');
         importServerResource.name = path[path.length - 1];
         resourceParent.addResource(importServerResource);
+        importServerResource.iconClass = getIconClass(resource);
+        return importServerResource;
     } else {
         resourceParent.importResource = resource;
+        resourceParent.iconClass = getIconClass(resource);
+        return resourceParent;
+    }
+};
+
+const getIconClass = (resource) => {
```

Comment:
Can you just create a simple mapping of resource types to icons instead of this function.

Reply:
Done

Reply:
I didn't mean to use a Map here but just a plain object. Anyway, this will do the job as well